src/k2/module/k23si/client/k23si_client.cpp

```diff
@@ -77,7 +77,18 @@ void K2TxnHandle::makeHeartbeatTimer() {
     });
 }
 
-std::unique_ptr<dto::K23SIReadRequest> K2TxnHandle::makeReadRequest(const dto::SKVRecord& record) const {
+std::unique_ptr<dto::K23SIReadRequest> K2TxnHandle::makeReadRequest(
+                        const dto::Key& key, const String& collection) const {
+    return std::make_unique<dto::K23SIReadRequest>(dto::K23SIReadRequest{
```

Comment:
you can avoid an extra move of the read request by using the emplace version here instead by just passing the ctor args to make_unique()